### ISSUE TITLE
Minion log in salt-ssh

### ISSUE THREAD

**Issue by anlutro** [saltstack/salt#25689]
Minion logging in salt-ssh doesn't work because the log directory doesn't exist:

```
vagrant@vagrant-test:~$ sudo python /tmp/.vagrant_740f1b_salt/salt-call --local -c /tmp/.vagrant_740f1b_salt -- test.ping
[WARNING ] Failed to open log file, do you have permission to write to /tmp/.vagrant_740f1b_salt/running_data/var/log/salt/minion?
local:
    True
vagrant@vagrant-test:~$ sudo mkdir -p /tmp/.vagrant_740f1b_salt/running_data/var/log/salt
vagrant@vagrant-test:~$ sudo python /tmp/.vagrant_740f1b_salt/salt-call --local -c /tmp/.vagrant_740f1b_salt -- test.ping
local:
    True
```

I was going to make a PR fixing this (simply creating the directory) but couldn't quite figure out where in the code it should be done.

**Comment by s0undt3ch** [comment]
Is it possible to include your versions report?

**Comment by anlutro** [comment]
Working off the latest 2015.8 head at the moment:

```
$ salt --versions-report
Salt Version:
           Salt: 2015.8.0rc2-3-g4121152
 
Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: Not Installed
         Python: 2.7.10 (default, Jul  1 2015, 10:54:53)
           RAET: Not Installed
        Tornado: 4.2
            ZMQ: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
       pycrypto: 2.6.1
 
System Versions:
           dist: debian stretch/sid 
        machine: x86_64
        release: 4.0.0-2-amd64
         system: debian stretch/sid
```

**Comment by s0undt3ch** [comment]
The problem started with https://github.com/saltstack/salt/commit/7fa1a9c8152df29a61a617dfbe3319aacdbc38b8

**Comment by anlutro** [comment]
You're right. That commit makes sense, so I guess reverting it is out of the question.

My initial idea was, in the SSH shim/client/whatever, to check if `os.path.dirname(log_path)` exists, and if not, create it.

**Comment by s0undt3ch** [comment]
I'm working on a simple fix.

**Comment by s0undt3ch** [comment]
This fix is against the 2015.5 branch but will be merge forwarded into 2015.8: https://github.com/saltstack/salt/pull/25694

**Comment by cachedout** [comment]
I merged @s0undt3ch 's #25694 so I'm going to go ahead and close this.
